pkg/actions/iam/create.go

```diff
@@ -0,0 +1,18 @@
+package iam
+
+import (
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+	"github.com/weaveworks/eksctl/pkg/kubernetes"
+)
+
+func (a *Manager) CreateIAMServiceAccount(iamServiceAccounts []*api.ClusterIAMServiceAccount, plan bool) error {
+	taskTree := a.stackManager.NewTasksToCreateIAMServiceAccounts(iamServiceAccounts, a.oidcManager, kubernetes.NewCachedClientSet(a.clientSet))
+	taskTree.PlanMode = plan
+
+	err := doTasks(taskTree)
+
+	cmdutils.LogPlanModeWarning(plan && len(iamServiceAccounts) > 0)
+
+	return err
+}
```

Comment:
I've left this here for now, if folks want me to remove this small refactor and put it into the next PR let me know.